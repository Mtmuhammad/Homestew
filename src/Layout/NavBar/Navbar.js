import React from "react";
import TopBar from "../TopBar/TopBar";
import "./Navbar.scss";
import { NavLink } from "react-router-dom";

const Navbar = () => {
  const handleCollapse = () => {
    let nav = document.querySelector("#navbarCollapse");
    let btn = document.querySelector(".navbar-toggler");
    nav.classList.remove("show");
    btn.classList.add("collapsed");
    btn.ariaExpanded = "false";
  };
  let activeClassName = "active nav-link";
  let inactiveClassName = "nav-link";
  return (
    <nav>
      <TopBar />
      {/* Navbar Main Logo */}
      <div className="d-flex justify-content-center">
        <div className="mt-3 mb-1">
          <a href="/">
            <img
              className="d-none d-lg-block logo"
              src="./homestewLogo.svg"
              alt="Homestew Logo"
            />
          </a>
        </div>
      </div>
      {/* end Navbar Main Logo */}

      {/* navbar */}
      <nav className="navbar navbar-expand-lg navbar-light">
        <div className="container-fluid">
          {/* navbar alt logo */}
          <a className="navbar-brand" href="/">
            <img
              className="d-lg-none logo-alt"
              src="/Homestew-alt-logo.svg"
              alt="Homestew Alt logo"
            />
          </a>
          {/* end navbar alt logo */}

          {/* navbar toggler */}
          <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarCollapse"
            aria-controls="navbarCollapse"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          {/* end navbar toggler */}

          {/* nav links */}
          <div
            className="justify-content-center collapse navbar-collapse"
            id="navbarCollapse"
          >
            <ul className="navbar-nav mb-2 mb-lg-0">
              <li className="nav-item">
                <NavLink
                end
                  reloadDocument
                  onClick={handleCollapse}
                  to="/"
                  className={({ isActive }) =>
                    isActive ? activeClassName : inactiveClassName
                  }
                >
                  HOME
                </NavLink>
              </li>

              <li className="nav-item">
                <NavLink
                  onClick={handleCollapse}
                  to="about"
                  className={({ isActive }) =>
                    isActive ? activeClassName : inactiveClassName
                  }
                >
                  ABOUT
                </NavLink>
              </li>

              <li className="nav-item">
                <NavLink
                  onClick={handleCollapse}
                  to="search"
                  className={({ isActive }) =>
                    isActive ? activeClassName : inactiveClassName
                  }
                >
                  RECIPES
                </NavLink>
              </li>

              <li className="nav-item">
                <NavLink
                  onClick={handleCollapse}
                  to="diets"
                  className={({ isActive }) =>
                    isActive ? activeClassName : inactiveClassName
                  }
                >
                  DIETS
                </NavLink>
              </li>

              <li className="nav-item">
                <NavLink
                  onClick={handleCollapse}
                  to="blog"
                  className={({ isActive }) =>
                    isActive ? activeClassName : inactiveClassName
                  }
                >
                  BLOG
                </NavLink>
              </li>

              <li className="nav-item">
                <NavLink
                  onClick={handleCollapse}
                  to="contact"
                  className={({ isActive }) =>
                    isActive ? activeClassName : inactiveClassName
                  }
                >
                  CONTACT
                </NavLink>
              </li>
            </ul>
          </div>
          {/* end nav links */}
        </div>
      </nav>
      {/* end navbar */}
      <hr />
    </nav>
  );
};

export default Navbar;
