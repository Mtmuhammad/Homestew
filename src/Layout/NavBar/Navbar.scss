@use "../../Styles/abstracts/mixins" as *;
@use "../../Styles/abstracts/variables" as *;


.logo {
  height: 130px;

  &-alt {
    margin-left: 15px;
    height: 40px;

    @include respond(sphone) {
      height: 35px;
    }
    @include respond(xsphone) {
      height: 30px;
    }
  }
}

.navbar-toggler {
  font-size: 2.25rem;
}

.dropdown-toggle::after {
  border: none;
}

.dropdown-toggle {
  text-align: center;

  & li {
    text-align: center;
  }
}

.dropdown-menu {
  padding: 1rem;
  background-color: $primary-green;
  font-size: 1.4rem;
  opacity: 0.9;

  & .dropdown-item {
    text-align: center;
    font-size: 1.5rem;

    &:active {
      background-color: $link-white;
      color: $primary-green;
    }
  }
}

.navbar-nav {
  @include respond(port) {
    background-color: rgba(128, 128, 128, 0.045);
  }
}

.nav-item {
  font-size: 2.1rem;
  margin-right: 10px;
  font-weight: bold;
  @include respond(port) {
    margin: 0 auto;
  }

  & > .active {
    &::after {
      content: "";
      border-bottom: 2px solid $primary-green;
      width: 100%;
      display: block;

      @include respond(port) {
        width: 0;
        border-bottom: none;
      }
    }
  }
}

.nav-link {
  position: relative;
  &.active {
    @include respond(port) {
      color: $primary-green !important;
    }
  }

  &::after {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    margin: auto;
    width: 0%;
    content: ".";
    color: transparent;
    background: $primary-green;
    height: 2px;
    transition: all 0.3s;
  }

  &:hover::after {
    width: 100%;
    @include respond(port) {
      width: 0;
    }
  }
}

hr {
  margin: 0;
  margin-top: 10px !important;
  border: 0;
  opacity: 1;
  background-color: #fff;
  border-top: 2px dotted #a8ba9a86;
}

@keyframes slideIn {
  0% {
    transform: translateY(1rem);
    opacity: 0;
  }

  100% {
    transform: translateY(0rem);
    opacity: 1;
  }

  0% {
    transform: translateY(1rem);
    opacity: 0;
  }
}

.slideIn {
  animation-name: slideIn;
}

.animate {
  animation-duration: 0.3s;
  animation-fill-mode: both;
}
