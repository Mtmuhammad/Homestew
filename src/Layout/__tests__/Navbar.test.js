import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Navbar from "../NavBar/Navbar";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
  });

  test("should match snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("render navbar brands", () => {
  test("should render main logo", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const mainLogo = screen.getByAltText(/homestew logo/i);
    expect(mainLogo).toBeInTheDocument();
    expect(mainLogo).toHaveClass("logo");
  });

  test("should render alternate logo on mobile", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const altLogo = screen.getByAltText(/homestew alt logo/i);
    expect(altLogo).toBeInTheDocument();
    expect(altLogo).toHaveClass("logo-alt");
  });
});

describe("navbar links", () => {
  test("should render home page link", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const homeLink = screen.getByText(/home/i);
    expect(homeLink).toHaveAttribute("href", "/");
  });
  test("should render about page link", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const aboutLink = screen.getByText(/about/i);
    expect(aboutLink).toHaveAttribute("href", "/about");
  });
  test("should render recipes page link", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const recipeLink = screen.getByText(/recipes/i);
    expect(recipeLink).toHaveAttribute("href", "/search");
  });
  test("should render diet page link", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const dietLink = screen.getByText(/diets/i);
    expect(dietLink).toHaveAttribute("href", "/diets");
  });
  test("should render blog page link", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const blogLink = screen.getByText(/blog/i);
    expect(blogLink).toHaveAttribute("href", "/blog");
  });
  test("should render contact page link", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const contactLink = screen.getByText(/contact/i);
    expect(contactLink).toHaveAttribute("href", "/contact");
  });
});

describe("should close navbar onclick", () => {
  test("home", () => {
    const handleCollapse = jest.fn();
    render(
      <BrowserRouter>
        <Navbar handleCollapse={handleCollapse} />
      </BrowserRouter>
    );
    window.innerWidth = 500;
    const homeLink = screen.getByText(/home/i);
    fireEvent.click(homeLink);
  });
  test("about", () => {
    const handleCollapse = jest.fn();
    render(
      <BrowserRouter>
        <Navbar handleCollapse={handleCollapse} />
      </BrowserRouter>
    );
    window.innerWidth = 500;
    const aboutLink = screen.getByText(/about/i);
    fireEvent.click(aboutLink);
  });
  test("recipes", () => {
    const handleCollapse = jest.fn();
    render(
      <BrowserRouter>
        <Navbar handleCollapse={handleCollapse} />
      </BrowserRouter>
    );
    window.innerWidth = 500;
    const recipesLink = screen.getByText(/recipes/i);
    fireEvent.click(recipesLink);
  });
  test("diets", () => {
    const handleCollapse = jest.fn();
    render(
      <BrowserRouter>
        <Navbar handleCollapse={handleCollapse} />
      </BrowserRouter>
    );
    window.innerWidth = 500;
    const dietsLink = screen.getByText(/diets/i);
    fireEvent.click(dietsLink);
  });
  test("blog", () => {
    const handleCollapse = jest.fn();
    render(
      <BrowserRouter>
        <Navbar handleCollapse={handleCollapse} />
      </BrowserRouter>
    );
    window.innerWidth = 500;
    const blogLink = screen.getByText(/blog/i);
    fireEvent.click(blogLink);
  });
  test("contact", () => {
    const handleCollapse = jest.fn();
    render(
      <BrowserRouter>
        <Navbar handleCollapse={handleCollapse} />
      </BrowserRouter>
    );
    window.innerWidth = 500;
    const contactLink = screen.getByText(/contact/i);
    fireEvent.click(contactLink);
  });
});
