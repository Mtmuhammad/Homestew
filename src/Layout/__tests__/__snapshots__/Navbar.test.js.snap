// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <nav>
    <div
      class="topbar"
    >
      <div
        class="container d-flex justify-content-between"
      >
        <div
          class="socials"
          style="margin-left: 15px;"
        >
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-twitter"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-solid fa-rss"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-facebook"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-pinterest"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-instagram"
            />
          </a>
          <a
            data-testid="social-icon"
            href="#0"
          >
            <i
              class="fa-brands fa-youtube"
            />
          </a>
        </div>
        <div
          class="socials"
        >
          <a
            data-testid="user-icon"
            href="/search"
          >
            <i
              class="fa-solid fa-magnifying-glass"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <div
        class="mt-3 mb-1"
      >
        <a
          href="/"
        >
          <img
            alt="Homestew Logo"
            class="d-none d-lg-block logo"
            src="./homestewLogo.svg"
          />
        </a>
      </div>
    </div>
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="Homestew Alt logo"
            class="d-lg-none logo-alt"
            src="/Homestew-alt-logo.svg"
          />
        </a>
        <button
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarCollapse"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="justify-content-center collapse navbar-collapse"
          id="navbarCollapse"
        >
          <ul
            class="navbar-nav mb-2 mb-lg-0"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="active nav-link"
                href="/"
              >
                HOME
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/about"
              >
                ABOUT
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/search"
              >
                RECIPES
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/diets"
              >
                DIETS
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/blog"
              >
                BLOG
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/contact"
              >
                CONTACT
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <hr />
  </nav>
</DocumentFragment>
`;
